winml/lib/Api/StringData.cpp

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "impl/StringData.h"
+
+namespace _winml {
+
+string_data::string_data(size_t size) :
+    buffer_(size) {}
+
+std::shared_ptr<_winml::idata> string_data::create(size_t size) {
+  return std::make_shared<string_data>(size);
+}
+
+size_t string_data::num_elements() {
+  return buffer_.size();
+}
+
+size_t string_data::size_in_bytes() {
+  WINML_THROW_HR(E_UNEXPECTED);
+}
+
+size_t string_data::num_buffers() {
+  return 1;
+}
+
+bool string_data::flush() {
+  // Vacuously true
+  return true;
+}
+
+std::vector<wss::IBuffer>& string_data::buffers() {
+  WINML_THROW_HR(E_UNEXPECTED);
+}
+
+gsl::span<byte> string_data::buffer(bool /*should_sync_buffer*/) {
+  return gsl::span<byte>(reinterpret_cast<byte*>(buffer_.data()), buffer_.size());
+}
+
+void string_data::set(size_t num_elements, const std::string_view* data) {
+  WINML_THROW_HR_IF_FALSE_MSG(
+      E_INVALIDARG,
+      num_elements <= buffer_.size(),
```

Comment:
Is this overflow check inverted?

Reply:
Nevermind, I see the IF_FALSE now

Reply:
The data's size is expected to be less than or equal to the underlying buffer.
If it is bigger, it isnt possible to set it - and the THROW is triggered.